Assets/MRTK/SDK/Editor/Inspectors/UX/StateVisualizer/StateVisualizerInspector.cs

```diff
@@ -0,0 +1,402 @@
+                animatableProperty = (AnimatableProperty)EditorGUILayout.EnumPopup(animatableProperty);
+
+                if (GUILayout.Button("Add the " + animatableProperty.ToString() + " Animatable Property"))
+                {
+                    CreateAnimatablePropertyInstance(stateContainerName, animatableProperty.ToString(), animationTargetIndex);
+
+                    serializedObject.ApplyModifiedProperties();
+                }
+            }
+        }
+
+
+        private void RenderAnimatableProperty(SerializedProperty animatableProperty, int animationTargetIndex)
+        {
+            SerializedProperty animatablePropertyName = animatableProperty.FindPropertyRelative("animatablePropertyName");
+            SerializedProperty stateName = animatableProperty.FindPropertyRelative("stateName");
+            SerializedProperty targetObj = animatableProperty.FindPropertyRelative("target");
+
+            GameObject targetGameObject = targetObj.objectReferenceValue as GameObject;
+
+            if (targetGameObject != null)
+            {
+                using (var check = new EditorGUI.ChangeCheckScope())
+                {
+                    EditorGUILayout.PropertyField(animatableProperty, true);
+
+                    if (check.changed)
+                    {
+                        instance.SetKeyFrames(stateName.stringValue, animationTargetIndex);
+                    }
+                }
+            }
+        }
+
+        private void RemoveKeyFrames(string stateName, string animatablePropertyName, int animationTargetIndex)
+        {
+            instance.RemoveKeyFrames(stateName, animationTargetIndex, animatablePropertyName);
+        }
+
+        private void CreateAnimatablePropertyInstance(string stateName, string propertyName, int animationTargetIndex)
+        {
+            instance.CreateAnimatablePropertyInstance(animationTargetIndex, propertyName, stateName);
+        }
+
+        private void RenderSetStateMachineButton()
+        {
+            if (GUILayout.Button("Generate New Animation Clips"))
+            {
+                instance.InitializeAnimatorControllerAsset();
+            }
+        }
+
+        private void RenderSyncStatesButton()
+        {
+            if (GUILayout.Button("Sync States with Interactive Element"))
+            {
+                instance.UpdateStateContainerStates();
+            }
+        }
+
+        private void RenderEndingButtons()
+        {
+            EditorGUILayout.Space();
+            EditorGUILayout.Space();
+
+            using (new EditorGUILayout.HorizontalScope())
+            {
+                RenderSetStateMachineButton();
+
+                if (instance.EditorAnimatorController != null)
+                {
+                    RenderSyncStatesButton();
+                }
+            }
+        }
+
+        // A target game object is one that is itself or a child of the root
+        private bool IsTargetObjectValid(GameObject targetObj)
+        {
```

Comment:
It looks like this method is just checking if a StateVisualizerComponent is present on this object, or on an ancestor. It looks like you can actually reuse `TransformExtensions.FindAncestorComponent<StateVisualizer>(this Transform startTransform, includeSelf=true)` rather than needing to re-implement this method. 

example of how you would use it: `return targetObj.transform.FindAncestorComponent<StateVisualizer>(true);`

Also, I noticed that you have the same method in another .cs file in this PR. Consider trying to factor that method out into one place to avoid code duplication.